### PR TITLE
[fix] Changes to support actual Ralph ClickHouse implementation

### PR DESCRIPTION
**Pull request by bmtcril**

### CODE REVIEW

File: .github/workflows/upgrade-python-requirements.yml [hunk 1]
```diff
@@ -10,14 +10,14 @@ on:
         required: true
         # If copying this template manually, you must provide your default branch name
         # in quotes, such as 'master'
-        default: "main"
```
Comment (bmtcril) [inline .github/workflows/upgrade-python-requirements.yml:13]: Changes unrelated, but I apparently needed to fix this workflow

File: xapi_db_load/backends/clickhouse_lake.py [hunk 2]
```diff
@@ -39,34 +43,72 @@ def create_db(self):
...
+
         sql = f"""
             CREATE TABLE IF NOT EXISTS {self.event_table_name} (
                 event_id UUID NOT NULL,
                 verb_id String NOT NULL,
                 actor_id UUID NOT NULL,
                 org String NOT NULL,
                 course_id String NOT NULL,
-                emission_time timestamp NOT NULL,
+                emission_time DateTime64(6) NOT NULL,
```
Comment (bmtcril) [inline xapi_db_load/backends/clickhouse_lake.py:74]: This is actually a super important change. By default ClickHouse will round datetimes to the second, this allows it to store the timestamps in the same way that Python sends them and prevents issues with comparing times.

**Review by pomegranited — APPROVED** [review]
:+1: This is working great, thank you @bmtcril !

Should we just recreate the `0.1` tag once this merges, since no one else is using this repo yet?

- [x] I tested this by rebuilding my tutor dev devstack from scratch, and ensured that the Superset custom dataset fields and dashboards installed by https://github.com/openedx/tutor-contrib-oars/pull/3 still work as expected.
   - installed and enabled [tutor-contrib-superset](https://github.com/openedx/tutor-contrib-superset)` plugin
   - installed and enabled the [tutor-contrib-oars#3](https://github.com/openedx/tutor-contrib-oars/pull/3) plugin from that PR's branch
   - installed and enabled [tutor-contrib-clickhouse](https://github.com/openedx/tutor-contrib-clickhouse) plugin, with a small modification to use this branch on init:
   ```diff
   diff --git a/tutorclickhouse/templates/clickhouse/tasks/demo-xapi.sh b/tutorclickhouse/templates/clickhouse/tasks/demo-xapi.sh
   index 177566d..525b4ec 100644
   --- a/tutorclickhouse/templates/clickhouse/tasks/demo-xapi.sh
   +++ b/tutorclickhouse/templates/clickhouse/tasks/demo-xapi.sh
   @@ -10,7 +10,7 @@ case "{{ CLICKHOUSE_LOAD_DEMO_DATA }}" in
            . virtualenv/bin/activate
   
           echo "Loading demo xAPI data..."
   -       pip install git+https://github.com/openedx/xapi-db-load@0.1#egg=xapi-db-load==0.1
  +       pip install git+https://github.com/openedx/xapi-db-load@bmtcril/fixes_for_new_ralph#egg=xapi-db-load==0.1
            xapi-db-load --backend clickhouse \
                --num_batches "{{ CLICKHOUSE_LOAD_DEMO_XAPI_BATCHES }}" \
                --batch_size "{{ CLICKHOUSE_LOAD_DEMO_XAPI_BATCH_SIZE }}" \
   ```
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A
- [x] <strike>I made sure any change in configuration variables is reflected in the corresponding
    client's `configuration-secure` repository.</strike> N/A

File: logs/ralph_clickhouse_100M_3/ralph_clickhouse_100m_10kbatch.txt [hunk 3]
```diff
@@ -0,0 +1,5399 @@
+Database password:Tables dropped
```
[the remark this answers is not in the document]
Comment (bmtcril) [reply logs/ralph_clickhouse_100M_3/ralph_clickhouse_100m_10kbatch.txt:1]: I want to remove them, but it seemed weird to have all of the tests _except_ for the one against the actual production-ish code stored for posterity. I will yank them next go round, though!

**Comment by bmtcril** [comment]
@pomegranited I've recreated the 0.1 tag with a release, in the future these will change over to the "v0.2" style names from our conventions but I didn't want to break your current workflow.

**Pull request merged**